modules/apps/depot/depot-test/src/testIntegration/java/com/liferay/depot/web/internal/search/spi/searcher/test/DepotSearchRequestContributorTest.java

```diff
@@ -0,0 +1,273 @@
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.depot.model.DepotEntry;
+import com.liferay.depot.service.DepotEntryGroupRelLocalService;
+import com.liferay.depot.service.DepotEntryLocalService;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.search.SearchContext;
+import com.liferay.portal.kernel.service.LayoutLocalService;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.search.searcher.SearchRequest;
+import com.liferay.portal.search.searcher.SearchRequestBuilder;
+import com.liferay.portal.search.searcher.SearchRequestBuilderFactory;
+import com.liferay.portal.search.spi.searcher.SearchRequestContributor;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Gustavo Lima
+ */
+@RunWith(Arquillian.class)
+public class DepotSearchRequestContributorTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group1 = GroupTestUtil.addGroup();
+		_group2 = GroupTestUtil.addGroup();
+	}
+
+	@Test
+	public void testContributeWithConnectedGroupId() throws Exception {
+		DepotEntry depotEntry = _addDepotEntry();
+
+		_depotEntryGroupRelLocalService.addDepotEntryGroupRel(
+			depotEntry.getDepotEntryId(), _group1.getGroupId());
+
+		SearchRequestBuilder searchRequestBuilder =
+			_searchRequestBuilderFactory.builder();
+
+		searchRequestBuilder.groupIds(_group1.getGroupId());
+
+		SearchRequest searchRequest = searchRequestBuilder.build();
+
+		_depotSearchRequestContributor.contribute(searchRequest);
+
+		searchRequestBuilder = _searchRequestBuilderFactory.builder(
+			searchRequest);
+
+		SearchContext searchContext = searchRequestBuilder.withSearchContextGet(
+			Function.identity());
+
+		Assert.assertEquals(
+			Arrays.toString(searchContext.getGroupIds()),
+			Arrays.toString(
+				new long[] {_group1.getGroupId(), depotEntry.getGroupId()}),
+			Arrays.toString(searchContext.getGroupIds()));
+	}
+
+	@Test
+	public void testContributeWithConnectedGroupIdToSeveralDepotEntries()
```

Comment:
Can you please check the naming of the method, because I'm not I understand the "To" there. Maybe `testContributeWithConnectedGroupIdWithSeveralDepotEntries`?

Reply:
Actually I just copied from the previous but I can rename